lib/src/boollist.dart

```diff
@@ -0,0 +1,252 @@
+    RangeError.checkValidRange(start, end, _length);
+    fill ??= false;
+
+    var startWord = start >> _entryShift;
+    var endWord = (end - 1) >> _entryShift;
+
+    var startBit = start & _entrySignBitIndex;
+    var endBit = (end - 1) & _entrySignBitIndex;
+
+    if (startWord < endWord) {
+      if (fill) {
+        _data[startWord] |= -1 << startBit;
+        _data.fillRange(startWord + 1, endWord, -1);
+        _data[endWord] |= (1 << (endBit + 1)) - 1;
+      } else {
+        _data[startWord] &= (1 << startBit) - 1;
+        _data.fillRange(startWord + 1, endWord, 0);
+        _data[endWord] &= -1 << (endBit + 1);
+      }
+    } else {
+      if (fill) {
+        _data[startWord] |= ((1 << (endBit - startBit + 1)) - 1) << startBit;
+      } else {
+        _data[startWord] &= ((1 << startBit) - 1) | (-1 << (endBit + 1));
+      }
+    }
+  }
+
+  /// Returns custom iterator for [BoolList].
+  ///
+  /// To provide null safety [Iterator.current] getter of returned iterator
+  /// returns `false` before and after iteration process.
+  @override
+  Iterator<bool> get iterator => _BoolListIterator(this);
+
+  void _setBit(int index, bool val) {
+    if (val) {
+      _data[index >> _entryShift] |= 1 << (index & _entrySignBitIndex);
+    } else {
+      _data[index >> _entryShift] &= ~(1 << (index & _entrySignBitIndex));
+    }
+  }
+
+  static int _lengthInWords(int bitsLength) {
+    return (bitsLength + (_bitsPerEntry - 1)) >> _entryShift;
+  }
+}
+
+class _GrowableBoolList extends BoolList {
+  static const int _growthFactor = 2;
+
+  _GrowableBoolList(int length)
+      : super._(
+          Uint32List(BoolList._lengthInWords(length * _growthFactor)),
+          length,
+        );
+
+  @override
+  set length(int length) {
+    RangeError.checkNotNegative(length, 'length');
+    if (length > _length) {
+      _expand(length);
+    } else if (length < _length) {
+      _shrink(length);
+    }
+  }
+
+  void _expand(int length) {
+    if (length > _data.length * BoolList._bitsPerEntry) {
+      _data = Uint32List(
+        BoolList._lengthInWords(length * _growthFactor),
+      )..setAll(0, _data);
+    }
+    _length = length;
+  }
+
+  void _shrink(int length) {
+    if (length < _length ~/ _growthFactor) {
+      var newDataLength = BoolList._lengthInWords(length);
+      _data = Uint32List(newDataLength)..setRange(0, newDataLength, _data);
+    }
+
+    for (var i = length; i < _data.length * BoolList._bitsPerEntry; i++) {
+      _setBit(i, false);
+    }
+
+    _length = length;
+  }
+}
+
+class _NonGrowableBoolList extends BoolList with NonGrowableListMixin<bool> {
+  _NonGrowableBoolList(int length)
+      : super._(
+          Uint32List(BoolList._lengthInWords(length)),
+          length,
+        );
+}
+
+class _BoolListIterator implements Iterator<bool> {
+  bool _current = false;
+  int _pos = 0;
+  final int _length;
+  int _wordIndex = -1;
+  int _bitIndex = BoolList._entrySignBitIndex;
+
+  int _mask;
```

Comment:
Should be initialized. Just set it to 0 or 1. I know it will be set to `1` in the first moveNext call, but it's better to tell the compiler that it won't ever be `null`.